pkg/controllers/repositoryserver/server.go

```diff
@@ -0,0 +1,207 @@
+	existingUserHostList := sets.String{}
+	for _, userProfile := range userProfiles {
+		existingUserHostList.Insert(userProfile.Username)
+	}
+
+	userAccess := h.RepositoryServerSecrets.serverUserAccess.Data
+	for username, password := range userAccess {
+		if existingUserHostList.Has(username) {
+
+			h.Logger.Info("Useralready exists, updating passphrase", username)
+			// Update password for the existing user
+			cmd := command.ServerSetUser(
+				command.ServerSetUserCommandArgs{
+					CommandArgs: &command.CommandArgs{
+						RepoPassword:   repoPassword,
+						ConfigFilePath: command.DefaultConfigFilePath,
+						LogDirectory:   command.DefaultLogDirectory,
+					},
+					NewUsername:  username,
+					UserPassword: string(password),
+				})
+			stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+			format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+			format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+			if err != nil {
+				errors.Wrap(err, "Failed to update existing user passphrase from the Kopia API server")
+			}
+			continue
+		}
+		cmd := command.ServerAddUser(
+			command.ServerAddUserCommandArgs{
+				CommandArgs: &command.CommandArgs{
+					RepoPassword:   repoPassword,
+					ConfigFilePath: command.DefaultConfigFilePath,
+					LogDirectory:   command.DefaultLogDirectory,
+				},
+				NewUsername:  username,
+				UserPassword: string(password),
+			})
+		stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+		format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+		format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+		if err != nil {
+			return errors.Wrap(err, "Failed to add new user to the Kopia API server")
+		}
+	}
+	repoServerAddress, err := GetPodAddress(context.Background(), h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get repository server address")
+	}
+	serverAdminUsername := string(h.RepositoryServerSecrets.serverAdmin.Data[serverAdminUserNameKey])
+	serverAdminPassword := string(h.RepositoryServerSecrets.serverAdmin.Data[serverAdminPasswordKey])
+
+	err = h.refreshServer(context.Background(), repoServerAddress, serverAdminUsername, serverAdminPassword)
+	if err != nil {
+		return errors.Wrap(err, "Failed to refresh repository server")
+	}
+	return nil
+}
+
+func (h *RepoServerHandler) refreshServer(ctx context.Context, serverAddress, username, password string) error {
+	repoPassword := string(h.RepositoryServerSecrets.repositoryPassword.Data[repoPasswordKey])
```

Comment:
Here as well, I think we won't need repoPassword.

Reply:
Same question

Reply:
Discussed here https://github.com/kanisterio/kanister/pull/1653#discussion_r1119730369